### ISSUE TITLE
Why is Backpack trying to query the database on route:list command ?

### ISSUE THREAD

**Issue by rabol** [Laravel-Backpack/CRUD#3565]
# Bug report

### What I did
Execute php artisan route:list

I use Homestead so normally I log on to that via homestead ssh, but sometimes I just execute simple commands from my terminal on my Mac.
And when i run in the terminal, then I don't have access to the mysql db, and therefore I get this error.

The big question is: why does backback need to access the db or why is this code executed... what else is going on ?

### What I expected to happen

get a list of routes in my app

### What happened
I found out that the for some reason backpack interact a little intrusive

in __construct on ResetPasswordController, this is called
        if (! backpack_users_have_email()) {
            abort(501, trans('backpack::base.no_email_column'));
        }

But why does it get to that in the console ?

### What I've already tried to fix it

??

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
PHP 8.0.0 (cli) (built: Nov 27 2020 12:26:05) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.0-dev, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.0, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
v8.29.0@d2eba352b3b3a3c515b18c5726b373fe5026733e

### BACKPACK VERSION:
4.1.36@779364646e3b57258b981ff9f1bdcd9887143836

**Comment by pxpm** [comment]
@rabol Like I told you on gitter, if you are building the controller `ResetPasswordController` is normal anything inside `__construct()` would run. 

If you run `php artisan migrate`, or `php artisan make:model` or whatever command that does not initialize controllers  you will not hit that, but `php artisan route:list` (and I think it's the only) will build the controllers thus hit the `__construct()` of that Controller that subsequently run:

```php
function backpack_users_have_email()
    {
        $user_model_fqn = config('backpack.base.user_model_fqn');
        $user = new $user_model_fqn();

        return \Schema::hasColumn($user->getTable(), 'email');
    }
```

Basically just checks if the `email` column exists in `user` table, the fact that it checks in console too is because of controller initialization like I told you before. 

Do you still think this is a bug ? 

Best,
Pedro

**Comment by rabol** [comment]
Imagine that you have a blog and use bp as backed. 
100.000 page view and 100.000 round trip to the db for no reason 

yes I think it is a buy. The check for email field should not be performed before a request for reset is issued

**Comment by pxpm** [comment]
No @rabol . That would not happen, unless your 100.000 request are for resetting password :) 

ONLY when that controller is initialized, that code is run. In a `production deployed` application, where users interact with the web interface, the only place where that code would run, is when the users go request a password reset. 

Just a note for future me: I suggested in gitter that we might be able to encapsulate this in a condition that checks for app running in console.

```php
if (!app()->runningInConsole()) {
//run the command
}
```

@promatik do you think this check is anytime needed in console scripts ? I think we are safe to assume that this should only run when app is not in console ? 

Best,
Pedro

**Comment by rabol** [comment]
aha... so it's only being used when the controller is in the routes file.

but why even check if there is a email column on the User model ?

the select of the user will simply return null, or ?

**Comment by pxpm** [comment]
We don't select any user, we just instantiate the `User` model to get the table name, and check if that table contains the `email` column.

We would probably remove that in 4.2 because we introduced the `setup_password_recovery_routes`, so at this time, we can probably remove it, but it will be a breaking change.

**Comment by promatik** [comment]
@pxpm I agree, I think that check is not necessary on console.
Looking at this, I wonder why the check is being made directly on database and not on a config 😅

> _We would probably remove that in 4.2 because we introduced the setup_password_recovery_routes, so at this time, we can probably remove it, but it will be a breaking change._

This is probably the way 👌
